### ISSUE TITLE
Setting deterministic lockstep changes the time step for the physics engine

### ISSUE THREAD

**Issue by treefort** [BabylonJS/Babylon.js#3293]
Check out the deterministic lockstep PG: https://www.babylonjs-playground.com/#DU4FPJ#3.

Changing the value for `lockstepMaxSteps`  affects the speed of the physics simulation.

**Comment by sebavanmicrosoft** [comment]
Ping @RaananW

**Comment by abow** [comment]
That PG gets faster and faster on each run even without changing lockstepMaxSteps.

**Comment by RaananW** [comment]
The PR gets faster due to a problem with creating an engine and deleting it. This addresses the issue - https://github.com/BabylonJS/Babylon.js/pull/3300

**Comment by treefort** [comment]
Are these two distinct issues? Running locally (without deleting the engine), I'm still seeing a direct correlation between render speed and `lockstepMaxSteps`.

**Comment by RaananW** [comment]
Yep, those are two distinct issues. I already found an error in the calculation of the deterministic step (will probably commit tomorrow), I just wanted to be sure we can test it correctly.

**Comment by RaananW** [comment]
I have pushed a fix. I am pasting my explanation from the PR here just for reference:

We should notice how deterministic step is working, and how we deal with max steps. If the scene is running at 20 FPS and we set the nax step to 3 and up, the scene will run in "simulated" 60 fps (as the max FPS target is 60 fps). if, however, we will set it to 1 or 2, the simulated fps will be lower (20 or 40 respectively).
